### PR TITLE
pod: Support new CRI-O namespaces

### LINKED ISSUE #960: 2.1.9 does not work with the latest CRI-O
CRI-O renamed the pod annotations and thus Clear Containers do not run properly with it anymore.

### PR DESCRIPTION
**Pull request by sameo**
The CRI-O code base went through renaming. All "ocid" instances
were replaced with "cri-o", including the annotation namespaces.
We fixed this and added proper annotations namespaces for CRI-O,
prefixed with io.kubernetes.cri-o.

This patch tries makes our current code base support both the new
and the legacy namespaces.

Fixes #960

Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>

### CODE REVIEW

File: src/pod.c [hunk 1]
```diff
@@ -105,6 +119,31 @@ add_container_mount(struct cc_oci_config *config) {
...
 
+/**
+ * Returns a pod namespace ID from an OCI annotation.
+ *
+ * \param annotation \ref oci_cfg_annotation.
+ *
+ * \return a valid pod namespace ID on success, and CC_POD_INVALID on failure.
+ */
+static enum pod_namespace_id
+pod_namespace_present(struct oci_cfg_annotation *annotation)
```
Comment (jodh-intel) [inline src/pod.c]: Missing `annotation != NULL && annotation->key != NULL` checks.

Comment (jodh-intel) [inline src/pod.c]: Also, could you add a brief test to `tests/pod_test.c` for this new function?

**Comment by jodh-intel** [comment]
Travis is complaining I'm afraid due to a missing "Fixes #XXX" commit message.
/me whistles... ;)

**Comment by chavafg** [comment]
qa-passed

File: src/pod.c [hunk 2]
```diff
@@ -105,6 +120,35 @@ add_container_mount(struct cc_oci_config *config) {
...
+{
+	if (annotation == NULL || annotation->key == NULL) {
+		return CC_POD_INVALID;
+	}
+
+	/* We only handle CRI-O and ocid annotations for now */
+	/* Let's check for CRI-O first */
+	if (strncmp(annotation->key, CC_POD_CRIO_NAMESPACE,
+		    CC_POD_CRIO_NAMESPACE_SIZE) == 0) {
```
Comment (jcvenegas) [inline src/pod.c]: that macro is a string literal. What about to use sizeof(CC_POD_CRIO_NAMESPACE) to get the length of that in the case that change in the future we wont forget change that value in the future?

**Comment by devimc** [comment]
LGTM

**Comment by jcvenegas** [comment]
lgtm

**Comment by chavafg** [comment]
qa-passed

**Pull request merged**